doozerlib/cli/images_streams.py

```diff
@@ -97,6 +97,13 @@ def images_streams_mirror(runtime, streams, only_if_missing, live_test_mode, dry
             else:
                 exectools.cmd_assert(cmd, retries=3, realtime=True)
 
+            if "aarch64" in runtime.arches:
+                arm_cmd = f'oc image mirror --filter-by-os linux/arm64 {brew_pullspec} {upstream_dest}-arm64'
```

Comment:
These images haven't been transformed to include the CI repos, which will have to be added as another layer in another build step on the arm build farm.
I therefore wonder whether it makes to suffix them with ´-art-arm64´ here, and use the ´-arm64´ suffix for the transformed (i.e. CI repo layer added) images?

Reply:
oh, we only need apply this on those items have transform value ? If true I can add a  judgment condition

Reply:
What images would be mirrored with this step? I think we only need the base (could probably use that without the transform), and the builder, which will require the transform layer

Reply:
according to the [request](https://issues.redhat.com/browse/ART-3279) and[ luke's comment ](https://issues.redhat.com/browse/ART-3279?focusedCommentId=19074383&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-19074383) I think I will just mirror builder and base with `-arm64` suffix.

Reply:
according to the request and[ luke's comment ](https://issues.redhat.com/browse/ART-3279?focusedCommentId=19074383&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-19074383) I think I will just mirror builder and base with `-arm64` suffix.

Reply:
@sosiouxme since we'll still need to apply the transforms to the images and commit them back to the registry, wdyt about using the `-art-arm64` suffix here to distinguish them from the `-arm64` images which would include the transform layer? Could you also point me towards the definition of that transform so I can add a build job to apply it on the Prow side?

Reply:
if you look at e.g. https://github.com/openshift/ocp-build-data/blob/f883a68216730db80bb49b5fd69f899861087f50/streams.yml#L30-L38 we have something of a convention around this already where we push to a name with `.art` on the end, and then there's a build to transform it into what actually gets used (which is what we put in our alignment PRs so CI will point at them).

so I think we can continue that here by simply adding the suffix. so e.g. 
`upstream_image_base: registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-{MAJOR}.{MINOR}.art`
- means we will push the ARM version to `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9.art-arm64`
- which will then require a transform into `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9-arm64`
- according to `upstream_image: registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-{MAJOR}.{MINOR}`

x86 CI is configured to use `registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9` and so arm64 CI should be configured to use that with the `-arm64` suffix (although maybe with the ARM64 cluster registry, will leave it to you how that last mile works).

sound good?

as far as the transforms... the transform in this example is
`transform: rhel-8/golang` which can be found in the [ci_transforms/rhel-8/golang](https://github.com/openshift/ocp-build-data/blob/openshift-4.9/ci_transforms/rhel-8/golang/) dir - it's just an openshift build with the base as the FROM, using that Dockerfile. I think that should translate directly into an ARM64 cluster build? I'm noticing though that the [base-repos transform](https://github.com/openshift/ocp-build-data/blob/openshift-4.9/ci_transforms/rhel-8/base-repos/Dockerfile) doesn't seem to be actually used any more, instead Test Platform is configured to add the repos some other way... maybe @jupierce could offer guidance?

Reply:
thanks for the details.
cc'ing @alvaroaleman @stevekuznetsov for their Prow expertise. Maybe you could sanity check this approach and point out any potential issues that you see with it, i.e. convert the transforms into buildconfigs running on a Prow cluster (specifically, arm01). Are we loosing any build or image-ref replacement automation with this?